File: v1/src/main/java/com/api/v1/medical_slots/CompletedMedicalSlotResponseDTO.java
package com.api.v1.medical_slots;

import com.api.v1.doctors.DoctorResponseDTO;
import com.api.v1.medical_slots.enums.MedicalSlotStatus;

import java.time.LocalDateTime;

public final class CompletedMedicalSlotResponseDTO extends MedicalSlotResponseDTO {

    private final LocalDateTime completedAt;

    private CompletedMedicalSlotResponseDTO(String id,
                                           MedicalSlotStatus status,
                                           DoctorResponseDTO doctor,
                                           LocalDateTime availableAt,
                                           LocalDateTime createdAt,
                                           LocalDateTime completedAt
    ) {
        super(id, status, doctor, availableAt, createdAt);
        this.completedAt = completedAt;
    }

    public static CompletedMedicalSlotResponseDTO from(MedicalSlot medicalSlot) {
        return new CompletedMedicalSlotResponseDTO(
                medicalSlot.id(),
                MedicalSlotStatus.COMPLETED,
                medicalSlot.toDTO().getDoctor(),
                medicalSlot.availableAt(),
                medicalSlot.createdAt(),
                medicalSlot.completedAt()
        );
    }

    public LocalDateTime getCompletedAt() {
        return completedAt;
    }
}

File: v2/src/main/kotlin/com/api/v2/medical_slots/CompletedMedicalSlotResponseDTO.kt
package com.api.v2.medical_slots

import com.api.v2.doctors.DoctorResponseDTO
import com.api.v2.medical_slots.enums.MedicalSlotStatus
import java.time.LocalDateTime

class CompletedMedicalSlotResponseDTO(
    id: String,
    doctor: DoctorResponseDTO,
    availableAt: LocalDateTime,
    createdAt: LocalDateTime,
    completedAt: LocalDateTime
): MedicalSlotResponseDTO(id, MedicalSlotStatus.COMPLETED, doctor, availableAt, createdAt)